packages/google_sign_in/google_sign_in_web/lib/src/utils.dart

```diff
@@ -74,10 +77,24 @@ GoogleSignInUserData? gisResponsesToUserData(
     id: payload['sub']! as String,
     displayName: payload['name'] as String?,
     photoUrl: payload['picture'] as String?,
-    idToken: credentialResponse.credential,
+    idToken: credentialResponse!.credential,
   );
 }
 
+/// Returns the expiration timestamp ('exp') of a [CredentialResponse].
+///
+/// May return `null` if the `credentialResponse` is null, its `credential`
+/// cannot be decoded, or the `exp` field is not set on the JWT payload.
+DateTime? getCredentialResponseExpirationTimestamp(
+    CredentialResponse? credentialResponse) {
+  final Map<String, Object?>? payload = getResponsePayload(credentialResponse);
+  if (payload == null || payload['exp'] == null) {
```

Comment:
How about:

```dart
final int? expiration = playload == null ? null : payload['exp'] as int?;
if (expiration == null) {
  return null;
}
```

so we don't need the force-unwrap?

Reply:
Good call, I also did a ternary for the `return`. Much shorter!